### PR TITLE
Feature/rep 103 crash by version

### PR DESCRIPTION
**Pull request by dovekie**
@alexanderyoung PTAL

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -554,6 +554,50 @@ def getDailyCrashes(appId,appName):
     return
 
 
+def getTrends(appId,appName):
+    """Get the number of daily app crashes for all versions of an app."""
```
Comment (alexanderyoung) [inline bin/critterget.py]: We should really give lists of args, raises, and returns.  Please add these to this and other functions you're introducing in this PR.

File: bin/critterget.py [hunk 2]
```diff
@@ -554,6 +554,50 @@ def getDailyCrashes(appId,appName):
...
+
+    getTopValues(appId,appName,trends_data)
+    getTimeseriesTrends(appId,appName,trends_data)
+
+
+def getTopValues(appId,appName,trendsData):
+    "Get top values from the trends endpoint"
+
+    trend_names = [u'appLoadsByVersion',
+                 u'crashesByVersion',
```
Comment (alexanderyoung) [inline bin/critterget.py]: Looks like this line is out of alignment with above.

File: bin/critterget.py [hunk 3]
```diff
@@ -554,6 +554,50 @@ def getDailyCrashes(appId,appName):
...
+                 u'appLoadsByOs',
+                 u'crashesByOs']
+
+    for trend in trend_names:
+        messages = u''
+        for version, value in trendsData[u'series'][trend][u'todayTopValues'].iteritems():
```
Comment (alexanderyoung) [inline bin/critterget.py]: Just an FYI here, but if you're defining a string without variable content and it doesn't have special chrs in it, no reason to use the 'u'.

File: bin/critterget.py [hunk 4]
```diff
@@ -601,7 +645,7 @@ def getCredentials(sessionKey):
             if auth is None:
                 print u'{} MessageType="ApteligentDebug" No credentials have been found for app {} . Maybe a setup issue?'.format(DATETIME_OF_RUN, myapp)
         retry += 1
-        time.sleep(5)
+        time.sleep(10)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Does the extra 5 seconds solve a problem or is this left over from testing?

File: bin/critterget.py [hunk 5]
```diff
@@ -632,40 +676,42 @@ def main():
...
-        getAPMGeo(key, apps[key], VOLUME, "ApmGeoVolume")
-        getAPMGeo(key, apps[key], ERRORS, "ApmGeoErrors")
-        getAPMGeo(key, apps[key], DATA, "ApmGeoData")
+        getTrends(key,apps[key]['name'])
+
+        getDailyAppLoads(key,apps[key]['name'])
+        getDailyCrashes(key,apps[key]['name'])
+        getCrashCounts(key,apps[key]['name'])
+
+        getGenericPerfMgmt(key,apps[key]['name'],"volume","device","DailyVolumeByDevice")
```
Comment (alexanderyoung) [inline bin/critterget.py]: We're using a mix of quotes here and below.  Let's use single quotes where possible please.

**Comment by alexanderyoung** [comment]
A few formatting and doc items, otherwise looks good.  Please fully test upgrade and new install paths before uploading to Splunk.

Thanks!

File: bin/critterget.py (hunk 4 above)
Comment (dovekie) [inline bin/critterget.py]: I tried a few values and settled on 10 secs as the best wait time to minimize the chance that the integration will fail to get its credentials during its first attempt and force me to wait 10 min before it tries again.

File: bin/critterget.py [hunk 6]
```diff
@@ -564,12 +569,18 @@ def getTrends(appId,appName):
 
 
 def getTopValues(appId,appName,trendsData):
```
Comment (alexanderyoung) [inline bin/critterget.py]: I should have caught this earlier.  Please single space the params per PEP 8

File: bin/critterget.py [hunk 7]
```diff
@@ -555,7 +555,12 @@ def getDailyCrashes(appId,appName):
 
 
 def getTrends(appId,appName):
-    """Get the number of daily app crashes for all versions of an app."""
+    """Calls the trends endpoint for an app
+
+    :param appId: (string) App ID used to request data from the API
```
Comment (alexanderyoung) [inline bin/critterget.py]: These look like intelij docstrings.  That's fine, but they look a little ugly to me.  The Google Python Style Guide is what I've used for more readable docstrings: https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

Up to you, just wanted to share what I use.

File: bin/critterget.py [hunk 8]
```diff
@@ -564,12 +569,18 @@ def getTrends(appId,appName):
...
-    trend_names = [u'appLoadsByVersion',
-                 u'crashesByVersion',
-                 u'appLoadsByOs',
-                 u'crashesByOs']
+    """Pulls todayTopValues data out of the trends dictionary for specific trends, and prints to Splunk.
+    :param appId: (string) App ID
+    :param appName: (string) Human-readable app name
+    :param trendsData: (dict) A dictionary of trends data
+    :return: None
+    :raises: KeyError if a trend does not exist in the trends dictionary (or if the trends data input was not a dictionary)
```
Comment (alexanderyoung) [inline bin/critterget.py]: You're actually handling KeyError, so this will never be raised.

File: bin/critterget.py [hunk 9]
```diff
@@ -585,11 +596,17 @@ def getTopValues(appId,appName,trendsData):
...
 def getTimeseriesTrends(appId,appName,trendsData):
-    "Get timeseries data from the trends endpoint"
-
-    for version in trendsData[u'series'][u'crashesByVersion'][u'categories'].keys():
+    """Pulls time series data out of the trends dictionary for crashes by version, and prints to Splunk.
+    :param appId: (string) App ID
+    :param appName: (string) Human-readable app name
+    :param trendsData: (dict) A dictionary of trends data
+    :return: None
+    :raises: KeyError if the data provided was not a dictionary
```
Comment (alexanderyoung) [inline bin/critterget.py]: Same comment above

**Comment by alexanderyoung** [comment]
Commented, LGTM once resolved.

**Pull request merged**